network/auth-header/src/index.ts

```diff
@@ -22,12 +23,23 @@ function getMaxParts (uris: string[]) {
   }, 0)
 }
 
-function getAuthHeaderByURI (authHeaders: Record<string, string>, maxParts: number, uri: string) {
+function getAuthHeaderByURI (authHeaders: Record<string, string>, maxParts: number, uri: string): string | undefined {
   const nerfed = nerfDart(uri)
   const parts = nerfed.split('/')
   for (let i = Math.min(parts.length, maxParts) - 1; i >= 3; i--) {
     const key = `${parts.slice(0, i).join('/')}/` // eslint-disable-line
     if (authHeaders[key]) return authHeaders[key]
   }
+  const urlWithoutPort = removePort(uri)
+  if (urlWithoutPort !== uri) {
+    return getAuthHeaderByURI(authHeaders, maxParts, urlWithoutPort)
+  }
   return undefined
 }
+
+function removePort (originalUrl: string) {
+  const urlObj = new URL(originalUrl)
+  if (urlObj.port === '') return originalUrl
+  const newUrlObj = new URL(`${urlObj.protocol}//${urlObj.hostname}${urlObj.pathname}${urlObj.search}${urlObj.hash}`)
```

Comment:
Is there a specific reason for not calling urlObj.port = “”? This line allocates a lot of string, and adds 200ns for just calling new URL.

Reply:
Yes, you are right. I'll release this patch ASAP https://github.com/pnpm/pnpm/pull/6717

Reply:
Thanks!